TSPL.docc/ReferenceManual/Expressions.md

```diff
@@ -1514,18 +1514,60 @@ Macro-expansion expressions have the following form:
 <#macro name#>(<#macro argument 1#>, <#macro argument 2#>)
 ```
 
-A macro-expansion expression omits the parentheses
+A macro-expansion expression omits the parentheses after the macro's name
 if the macro doesn't take any arguments.
 
-A macro expression can't appear as the default value for a parameter,
-except for the [`file()`][] and [`line()`][] macros from the Swift standard library.
+A macro-expansion expression can't appear as the default value for a parameter,
+except the [`file()`][] and [`line()`][] macros from the Swift standard library.
 When used as the default value of a function or method parameter,
-These macros' value is determined
-when the default value expression is evaluated at the call site.
+these macros are evaluated using the source code location of the call site,
+not the location where they appear in a function definition.
 
 [`file()`]: https://developer.apple.com/documentation/swift/file()
 [`line()`]: https://developer.apple.com/documentation/swift/line()
 
+You use macro expressions to call freestanding macros.
+To call an attached macro,
+use the custom attribute syntax described in <doc:Attributes>.
+Both freestanding and attached macros are expanded as follows:
+
+1. Swift parses the source code and performs type checking,
```

Comment:
We do type-check the macro arguments, but (especially with attached macros) not all of the ASTs that are provided to the macro implementation are type-checked, and the results of type checking are not currently available to macros. I think it might be clearer to drop the "performs type checking" part here, because macros work entirely in the syntactic space:

```suggestion
1. Swift parses the source code
```